src/state/Pointer.ts

```diff
@@ -57,15 +57,15 @@ export function walk(segments: string[], object: any, clone = true): PointerTarg
 	}, pointerTarget);
 }
 
-export class Pointer {
+export class Pointer<T = any, U = any> {
```

Comment:
There seems like there is a more direct relationship between a Store and a Pointer?

Reply:
Or at least a relationship between `T` and `U`?

Reply:
Ideally, but I'm not sure of a good way to express that. If you look at the `State` interface, where there is `P1 = keyof T`, `P2 = keyof T[P1]` etc. `U` might correspond to any of those signatures. We could have an interface similar to `State` that iterates all the possibilities, e.g.
```typescript
interface createPointer {
 <T, P1 extends keyof T>(p1: P1): Pointer<T, T[P1]>;
 <T, P1 extends keyof T, P2 extends keyof T[P1]>(p1: P1, p2: P2): Pointer<T, T[P1][P2]>;
 // ...
}
```

But even that's tricky. Because we aren't actually passing anything with type `T`, it would have to be explicitly typed. That ends up with all the subsequent types needing to be passed or just collapsing to their defaults if they're provided.

Reply:
👍